### PR TITLE
Implement #4984: Add option to highlight vomit, vandalism, litter and handymen

### PR DESCRIPTION
**Pull request by rwjuk**
This PR partially implements #4984, by adding an option to the drop-down 'view' menu that highlights vomit, litter, vandalism and handymen. This allows users to see where the problems in their park are and fix them. This is apparently a feature in RCTC.

Not sure whether this is a road we want to go down, but I was experimenting and thought it worth putting this up for discussion!

image

### CODE REVIEW

**Comment by mrtnptrs** [comment]
I actually really like this and find this a better implementation then the one RCTC provides: this viewing mode only shows the paths in grey, the broken benches in color, the vomit in color and the handyman to clean that up. 

The only thing that I'm missing/find a bit weird, is that when you you fix vandalism by replacing the broken benches and bins, they just dissapear in that viewing mode with no money-cost animation. And, as in RCTC, it would be really nice to be able to just click on the broken scenery to automaticly fix it instead of having to go to the scenery window. But as it is now already looks really good.

**Comment by AaronVanGeffen** [comment]
This is a neat addition! Thanks for taking the time to implement this.

I get that it's hiding rides and their supports for visibility, but could the water be shown, still? I find it a little disorientating when all the water is suddenly gone (cf. Diamond Heights).

That leads me to wonder how this would work with underground paths? I suppose it'd work in combination with the underground view, but I haven't tested this.

File: src/openrct2/paint/tile_element/Path.cpp [hunk 1]
```diff
@@ -695,6 +701,11 @@ void path_paint(paint_session * session, uint8 direction, uint16 height, rct_til
         }
     }
 
+    if (gCurrentViewportFlags & VIEWPORT_FLAG_HIGHLIGHT_PATH_ISSUES)
+    {
+        imageFlags = 0x21700000;
```
Comment (marijnvdwerf) [inline src/openrct2/paint/tile_element/Path.cpp]: Split this up into remap type and shifted colours/filters

Comment (AaronVanGeffen) [reply src/openrct2/paint/tile_element/Path.cpp]: NB: the same magic value is used in other places, when selecting scenery to save with a track design. [Fence.cpp line 170](https://github.com/OpenRCT2/OpenRCT2/pull/6996/files#diff-7f10c704c1b6b6a80b7bedad11219eb2R170) in particular stands out as it shows up in this PR.

**Comment by AaronVanGeffen** [comment]
I've taken the liberty to rebase this branch, as it had a few conflicts. With regards to the flag values, I've also split those into bitshifts. Do we already have named constants for these anywhere?

**Comment by rwjuk** [comment]
@AaronVanGeffen Thanks! (Got distracted with OpenLoco...) If we don't have them, we should figure them out and create some.

**Comment by marijnvdwerf** [comment]
The meaning of the colours generally are `c1 << 19 | c2 << 24 | flags << 29`.
Check the `SPRITE_ID_PALETTE_COLOUR_1` macros. (Though it might also be `IMAGE_TYPE_TRANSPARENT`, which doesn't yet seem to have a decent macro.

**Comment by AaronVanGeffen** [comment]
Thanks, @marijnvdwerf. I have replaced the constants with `SPRITE_ID_PALETTE_COLOUR_1(PALETTE_46)`.

**Comment by SidaRadis** [comment]
Would highlighting full litter bins be possible? They can be really hard to see with the default litter bin for example.

**Comment by AaronVanGeffen** [comment]
> Would highlighting full litter bins be possible? They can be really hard to see with the default litter bin for example.

It's a little more difficult due to the way the bins are stored, but it should be doable.

**Pull request merged**